src/printer.js

```diff
@@ -338,19 +338,28 @@ function genericPrintNoParens(path, options, print, args) {
         );
       }
 
-      // Avoid indenting sub-expressions in assignment/return/etc statements.
-      if (
-        parent.type === "AssignmentExpression" ||
-        parent.type === "VariableDeclarator" ||
-        shouldInlineLogicalExpression(n) ||
+      // Avoid indenting sub-expressions in some cases where the first sub-expression is already
+      // idented accordingly. We should ident sub-expressions where the first case isn't idented.
+      const shouldNotIdent =
         parent.type === "ReturnStatement" ||
         (parent.type === "JSXExpressionContainer" &&
           parentParent.type === "JSXAttribute") ||
         (n === parent.body && parent.type === "ArrowFunctionExpression") ||
         (n !== parent.body && parent.type === "ForStatement") ||
+        parent.type === "ConditionalExpression";
+
+      const shouldIdentIfInlining =
+        parent.type === "AssignmentExpression" ||
+        parent.type === "VariableDeclarator" ||
         parent.type === "ObjectProperty" ||
-        parent.type === "Property" ||
-        parent.type === "ConditionalExpression"
+        parent.type === "Property";
+
+      const logicalSubExpressionExists = n.left.type === "LogicalExpression";
```

Comment:
You don't use this variable anywhere; did you mean to use it to replace the `n.left/type !== n.type` [on line 361](https://github.com/prettier/prettier/pull/2770/files#diff-76f26752c3761490942405c0704a8e1dR361)?

Reply:
Oops my bad!!

Reply:
I'll update this. Are you ok with this name?